src/components/Article.js

```diff
@@ -72,7 +71,7 @@ const ArticleSection = ({
     }
   }
 
-  useArticleLinksNavigation('Article', lang, contentRef, linkHandlers, [page, textSize], { galleryItems, articleTitle, namespace, id })
+  useArticleLinksNavigation('Article', lang, contentRef, linkHandlers, [page, textSizeAdjustment], { galleryItems, articleTitle, namespace, id })
```

Comment:
Why don't you depend on the text size itself here instead of the "adjusted" variable. Typically, it's `useEffect`'s job to figure out if dependencies have changed between cycles.

Reply:
It’s because as `textSize` state changes instantly when the users plays with the slider, the `useEffect` in `useArticleLinksNavigation` triggers every time and consequently sets the softkeys at the wrong moment. Hence the `hasAdjusted` state to link `TextSize` and `Article`. We could also just have the text size slider update the preview text in `TextSize` only and not the article text on the background, that way we could use `textSize` as the dependency directly

Reply:
Right, I remember this issue now. And looking at the code, a clean solution would involve `useSoftkeys` understanding that it should update the handlers for a component that is not top-most at the moment. Looking at the SoftkeyReducer it seems possible but it adds scope when we should be in wrap up mode here.

Let's talk about this later.